File: boj/10845.cpp
#include <iostream>
#include <queue>


int n;
std::string str;
void solve();

int main()
{

    std::cin>>n;
    solve();
    return 0;
}


void solve()
{
    std::queue<int>q;
    int a;

    while(n--){

        std::cin>>str;
        if(str == "push") {
            std::cin>>a;
            q.push(a);
        }else if(str == "front"){
            if(q.size() == 0) { 
                std::cout<<"-1"<<std::endl;
            }
            else {
                std::cout<<q.front()<<std::endl;
            }

        }else if(str == "back"){
            if(q.size() == 0) { 
                std::cout<<"-1"<<std::endl;
            }
            else {
                std::cout<<q.back()<<std::endl;
            }

        }else if(str == "size"){
            std::cout<<q.size()<<std::endl;
        }else if(str == "empty"){
            if(q.empty() == true) std::cout<<1<<std::endl;
            else std::cout<<0<<std::endl;

        }else if(str == "pop"){
            if(q.size() == 0) { 
                std::cout<<"-1"<<std::endl;
            }
            else {
                std::cout<<q.front()<<std::endl;
                q.pop();
            }

        }else{

        }


    }

}

File: boj/10845.py
import sys

def run(n):
    q = []
    for _ in range(n):
        data = list(map(str,sys.stdin.readline().split()))
        if data[0] == 'push':
            q.append(int(data[1]))
        if data[0] == 'front':
            if len(q):
                print(q[0])
            else:
                print(-1)
        if data[0] == 'empty':
            if len(q):
                print(0)
            else:
                print(1)
        if data[0] == 'size':
            print(len(q))
        if data[0] == 'back':
            if len(q):
                print(q[-1])
            else:
                print(-1)
        if data[0] == 'pop':
            if len(q):
                print(q[0])
                del q[0]
            else:
                print(-1)


if __name__ == "__main__":
    n = int(input())
    run(n)